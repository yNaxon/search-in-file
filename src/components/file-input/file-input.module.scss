.fileInput {
  border-radius: .75rem;
  background-color: #d9d9d9;
  padding-block-start: .75rem;
  padding-block-end: .75rem;
  padding-inline-start: .75rem;
  padding-inline-end: .75rem;
  cursor: pointer;
  
  display: inline-block;

  &:hover {
    background-color: #bfbfbf;
  }

  &:focus-within {
    box-shadow: 0 0 0 2px inset orange;
  }
}

.input {
  visibility: hidden;
  height: 0;
  width: 0;
}
