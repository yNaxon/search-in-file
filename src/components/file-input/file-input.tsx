import { InputHTMLAttributes } from 'react';
import classNames from 'classnames';
import styles from './file-input.module.scss';

export function FileInput({
  className,
  type,
  placeholder = 'Select File',
  ...props
}: InputHTMLAttributes<HTMLInputElement>) {
  return (
    <label
      className={classNames(styles.fileInput, className)}
    >
      {placeholder}
      <input
        type="file"
        className={styles.input}
        {...props}
      />
    </label>
  )
}