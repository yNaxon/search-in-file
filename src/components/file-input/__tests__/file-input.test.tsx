import React from 'react';
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer';
import { FileInput } from '../file-input';

it('renders without crashing', () => {
  render(<FileInput placeholder="Upload File" />)
});

it('renders correctly', () => {
  const tree = renderer
    .create(<FileInput placeholder="Upload File" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});