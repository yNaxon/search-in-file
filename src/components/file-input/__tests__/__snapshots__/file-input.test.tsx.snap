// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<label
  className="fileInput"
  data-testid="file-input"
>
  Upload File
  <input
    className="input"
    type="file"
  />
</label>
`;
